.github/workflows/pull-request-checks.yml

```diff
@@ -12,9 +12,34 @@ concurrency:
 
 permissions:
     contents: read
+env:
+    CI_LABEL: "ci:approve-public-fork-ci"
 
 jobs:
+    check-run-approval:
+        permissions:
+            pull-requests: write
+            contents: read
+        # to prevent CI from running against public forks without us looking at the code, we will check for the presence of the proper label
+        runs-on: ubuntu-latest
+        steps:
+            -   name: "If the event is to synchronize, we should remove the label from the PR"
```

Comment:
So every time they update the PR we need to re-add the label

Reply:
Correct.  This prevents introducing malicious code after the initial PR is opened without our knowledge.